File: leetcode/algorithms/string/robot-return-to-origin.ts
/*
    leetcode
    category: "Algorithms"
    problem title: "Robot Return to Origin"
*/


function judgeCircle(moves: string): boolean {
    let x: number = 0,y: number = 0;
    
    for(let i: number = 0;i<moves.length;++i){
        switch(moves[i]){
            case "U":
                ++y;
                break;
            case "D":
                --y;
                break;
            case "L":
                --x;
                break;
            case "R":
                ++x;
                break;
        }
    }
    
    return x===0 && y===0;

};

File: leetcode/algorithms/string/robot-return-to-origin.go
func judgeCircle(moves string) bool {
	var verticalOffset int = 0
	var horizontalOffset int = 0

	var i int = 0
	for i < len(moves) {
		switch string(moves[i]) {
		case "U":
			verticalOffset++
			break
		case "D":
			verticalOffset--
			break
		case "L":
			horizontalOffset--
			break
		case "R":
			horizontalOffset++
			break
		}
		i++
	}

	return verticalOffset == 0 && horizontalOffset == 0

}